src/nrniv/CMakeLists.txt

```diff
@@ -304,11 +304,20 @@ if(NRN_USE_BACKWARD)
     PROPERTY COMPILE_DEFINITIONS USE_BACKWARD=1)
 endif()
 
-# NVHPC/21.7 cannot compile znorm.c with -O2 or above. See also:
-# https://forums.developer.nvidia.com/t/nvc-21-7-regression-internal-compiler-error-can-only-coerce-indirect-args/184847
-if(NRN_HAVE_NVHPC_COMPILER AND ${CMAKE_C_COMPILER_VERSION} VERSION_EQUAL 21.7)
-  set_source_files_properties(${PROJECT_SOURCE_DIR}/src/mesch/znorm.c PROPERTIES COMPILE_OPTIONS
-                                                                                 -Mnovect)
+if(NRN_HAVE_NVHPC_COMPILER)
+  # NVHPC/21.7 cannot compile znorm.c with -O2 or above. See also:
+  # https://forums.developer.nvidia.com/t/nvc-21-7-regression-internal-compiler-error-can-only-coerce-indirect-args/184847
+  if(${CMAKE_C_COMPILER_VERSION} VERSION_EQUAL 21.7)
+    set_source_files_properties(${PROJECT_SOURCE_DIR}/src/mesch/znorm.c PROPERTIES COMPILE_OPTIONS
+                                                                                   -Mnovect)
+  endif()
+  # For NVHPC we will rely on FE exceptions as opposed to errno in order to make use of faster
+  # builtins. One caveat is that if we use an optimization level greater than -O1, the FE exception
+  # is not raised. See https://github.com/neuronsimulator/nrn/pull/1930
+  set_source_files_properties(${PROJECT_SOURCE_DIR}/src/oc/math.cpp
+                              PROPERTIES COMPILE_DEFINITIONS NVHPC_CHECK_FE_EXCEPTIONS=1)
+  set_source_files_properties(${PROJECT_SOURCE_DIR}/src/oc/math.cpp PROPERTIES COMPILE_FLAGS -O1)
```

Comment:
`COMPILE_FLAGS` is apparently deprecated: https://cmake.org/cmake/help/latest/prop_tgt/COMPILE_FLAGS.html

Reply:
Indeed. We were discussing today about the flags, `-O1` comes in after `-O2`: 
```
[370/1313] nvc++ -DHAVE_CONFIG_H -DMPICH_SKIP_MPICXX=1 -DMPI_NO_CPPBIND=1 -DNRN_ENABLE_THREADS -DOMPI_SKIP_MPICXX=1 -DR123_USE_INTRIN_H=0 -Dnrniv_lib_EXPORTS -DNVHPC_CHECK_FE_EXCEPTIONS=1 -I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-I***-g  -O2  -fPIC --diag_suppress=1,47,111,128,170,174,177,180,186,301,541,550,816,941,2465 --c++17 -O1 -MD -MT src/nrniv/CMakeFiles/nrniv_lib.dir/__/oc/math.cpp.o -MF src/nrniv/CMakeFiles/nrniv_lib.dir/__/oc/math.cpp.o.d -o src/nrniv/CMakeFiles/nrniv_lib.dir/__/oc/math.cpp.o -c /gpfs/bbp.cscs.ch/home/savulesc/nrn/src/oc/math.cpp
nvc++-Info-Switch -Mvect forces -O2
```

Reply:
OK...I think the `nvc++-Info-Switch -Mvect forces -O2` message is related to having both different `-O` flags. But if this makes the tests pass then I guess it's OK...to test that then I suppose we should launch the CoreNEURON pipeline against this branch?

Reply:
Looks like so. See https://bbpgitlab.epfl.ch/hpc/coreneuron/-/pipelines/68549

Reply:
Ok so it works (there is an error in the nctx plasticity, but that is unrelated)